### PR TITLE
Add simple workflow to worskspaces container, and create default "workspaces" folder

### PR DESCRIPTION
**Pull request by pysailor**
As discussed earlier, the "workspaces" folder in the root should be created by default in PI, and it needs to be published

### CODE REVIEW

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/71/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/72/
Test PASSed.

File: src/ploneintranet/workspace/tests/test_content_type.py [hunk 1]
```diff
@@ -125,8 +125,8 @@ def test_add_workspacecontainer(self):
         api.content.create(
             self.portal,
             'ploneintranet.workspace.workspacecontainer',
-            'workspaces',
-            title='Workspaces'
+            'test-workspaces',
+            title='Test Workspaces'
         )
```
Comment (mattss) [inline src/ploneintranet/workspace/tests/test_content_type.py:130]: These tests no longer need to create a workspace container

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/78/
Test PASSed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/79/
Test PASSed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/80/
Test PASSed.

**Comment by pysailor** [comment]
@mattss @gyst I've reworked this PR. Instead of relying on the simple_publication_workflow (until now associated to workspacecontainer), we create our own simple workflow for the wokspaces container. IHMO this has the following advantages:
1. We don't need to create a separate permission à la "Add workspace" - we can use the canonical "Add portal content"
2. We will be able to more easily support scenarios where we have more than one workspaces container, with potentially different policies for creating workspaces 

For good measure, I gave 2 states to the new workflow:
- Open (default): any authenticated user can add a WS
- Restricted:any authenticated user can view, only Contributors can add a WS

The latter state can currently only be reached in code, not via the UI, but I added a test that shows this.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/81/
Test PASSed.

File: src/ploneintranet/workspace/profiles/default/workflows/ploneintranet_workspacecontainer_workflow/definition.xml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    acquired="False">
+      <permission-role>Authenticated</permission-role>
+    </permission-map>
+    <permission-map name="List folder contents"
+                    acquired="False">
+      <permission-role>Authenticated</permission-role>
+    </permission-map>
+    <permission-map name="Modify portal content"
+                    acquired="False">
+      <permission-role>Authenticated</permission-role>
```
Comment (gyst) [inline src/ploneintranet/workspace/profiles/default/workflows/ploneintranet_workspacecontainer_workflow/definition.xml]: This is a problem. We should not allow any authenticated user to modify the workspacecontainer. This should be removed and just acquire the sitewide ModifyPortalContent settings.

Additionally, only Manager, not even Site Manager, should be allowed to create or delete workspacecontainers. This requires a special permission. I've just fixed a comparable pattern in another site. The workspacecontainer is more like a site utility than normal content. The code relies on it being present. Deleting it is catastrophic. Adding another one is a big decision. Most site managers I'm working with cannot be trusted to not f*ck this up. They have, and they will, if you don't protect them against themselves.

**Comment by gyst** [comment]
FYI @pysailor I will schedule additional workspace security stories for the next sprint:
- Remove the View tweaks, instead create a new permission 'See workspace exists'
- A one-state workflow for folders and images within workspaces
- A special workspacesubfolder class (instead of a normal Folder) so we can lock-down addable types and ensure that any type that can added is mapped to a local workflow that blocks permission acquisition.

**Comment by pysailor** [comment]
@gyst Thanks for the feedback. If I fix the workflow definition like you wrote in your comment, would you then want to merge this PR? I don't see a strong connection to the other workspace security stories, so they could/should go into (a) separate PR(s) IMHO.

**Comment by gyst** [comment]
@pysailor Yes sure the extra stories are just FYI not for this pull.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/83/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/84/
Test PASSed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/85/
Test FAILed.

**Comment by pysailor** [comment]
jenkins retest this

**Comment by pysailor** [comment]
jenkins please retest this

**Comment by mattss** [comment]
Test this please jenkins

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/87/
Test PASSed.

**Comment by pysailor** [comment]
@gyst I removed the "Modify portal content" permission from the workflow (so that it gets acquired instead) and protected adding a workspace container with ManagePortal (which Site Manager does not have).

**Pull request merged**